### PR TITLE
Remove Elasticsearch 6.3 logic

### PR DESCRIPTION
**Pull request by pquentin**
Rally only supports Elasticsearch 6.8+.

### CODE REVIEW

**Review by michaelbaamonde — APPROVED** [review]
LGTM

File: esrally/driver/driver.py [hunk 1]
```diff
@@ -354,8 +354,7 @@ def _after_track_prepared(self):
         self.send(
             self.start_sender,
             PreparationComplete(
-                # older versions (pre 6.3.0) don't expose build_flavor because the only (implicit) flavor was "oss"
-                cluster_version.get("build_flavor", "oss"),
```
Comment (pquentin) [inline esrally/driver/driver.py:358]: This breaks the `ClusterEnvironmentInfo` telemetry device when no build flavor is set, eg. `esrally race --revision=latest --track=geonames --test-mode --car=4gheap,basic-license` breaks.

**Comment by pquentin** [comment]
Please take another look!

**Review by michaelbaamonde — APPROVED** [review]
LGTM!

**Pull request merged**